grails-app/services/org/pih/warehouse/inventory/RequisitionStatusTransitionEventService.groovy

```diff
@@ -13,12 +13,18 @@ import org.codehaus.groovy.grails.commons.GrailsApplication
 import org.pih.warehouse.core.MailService
 import org.pih.warehouse.core.Person
 import org.pih.warehouse.requisition.Requisition
+import org.pih.warehouse.requisition.RequisitionService
+import org.pih.warehouse.requisition.RequisitionStatusTransitionEvent
+import org.springframework.context.ApplicationEvent
+import org.springframework.context.ApplicationListener
 
+class RequisitionStatusTransitionEventService implements ApplicationListener<RequisitionStatusTransitionEvent> {
 
-class RequisitionStatusTransitionEventService {
+    static transactional = true
```

Comment:
Is this transactional really necessary? (I know that you marked it to remember adding a transaction in Grails 3, but I am not sure if there is a need for the transaction)

Reply:
If we're just doing reads then we don't need a transaction. If you're writing to the database then we need a transaction.

Reply:
In general I would not agree, because not having a transaction boundary also causes LazyInitializationExceptions while reading (assuming you don't use osiv). Anyway as I mentioned, it was actually added for debugging purposes and finding a solution for the Lazy that was thrown.
I will remove it.